valhalla/baldr/graphreader.h

```diff
@@ -695,6 +695,13 @@ class GraphReader {
     return edgeinfo(edgeid, NO_TILE);
   }
 
+  /**
+   * Get the shape of an edge
+   * @param edgeid
+   * @return the encoded shape (string) for specified id.
+   */
+  std::string edge_shape(const valhalla::baldr::GraphId& edgeid);
```

Comment:
Without going into details, does this qualify for either a `static` or `const` function? If its purely a stateless helper function, then it should be `static`. If it uses the object state but doesn't change it, then we can mark it a `const` function.

Reply:
it changes the state of graph_reader, thus can not be mark const.

Reply:
There is a hidden trick that inside this function, called GetGraph tile, which is not const.

Reply:
I see. `GetGraphTile` updates its `cache_` before returning the tile (it also locks the mutex, although that can be marked `mutable` without violating const-correctness).

Reply:
correct, in order to first get the tile that it needs to do the work it must mutate internal state.